final release/format_data.py

```diff
@@ -0,0 +1,208 @@
+
+# coding: utf-8
+
+# In[45]:
+
+
+import numpy as np
+import pickle
+import matplotlib.pyplot as plt
+import h5py
+from skimage.transform import resize
+from astropy.io import ascii
+#import ask_first
+from sklearn.utils import shuffle
+from keras.utils import np_utils
+from astropy.coordinates import SkyCoord
+import astropy.units as u
+
+import sys, os
+
+
+# In[46]:
+
+def get_data(first_path, second_path, dim = 256, seed = 0, conf_matrix = True, flip_train_test = False):
+    # Loading data
+    data = h5py.File(first_path, 'r')
+
+    images = np.asarray(data['images'])
+    labels = np.asarray(data['labels'])
+
+    # Opening Catalogues
+    asu = ascii.read(second_path).to_pandas()
+
+    # Opening Entire First Catalogue with ask_first script (Only for reference)
+    #first = ask_first.read_catalogue("catalog_14dec17.bin")
+
+
+    # Splitting classes
+    fri_images = images[np.where(labels == False)]
+    frii_images = images[np.where(labels == True)]
+
+    # Extracting data 
+    fri_data = data['fri_data']
+    frii_data = data['frii_data']
+
+    # Converting coordinates into a Astropy Skycoord readable format
+    frii_c = np.hstack((np.asarray(list(frii_data['_RA'])).reshape(-1, 1), np.asarray(list(frii_data['_DE'])).reshape(-1, 1)))
+    fri_c = np.hstack((np.asarray(list(fri_data['ra'])).astype(str).reshape((-1, 1)), np.asarray(list(fri_data['dec'])).astype(str).reshape((-1, 1))))
+    
+    # Coverting coordinates of wildly varying formats to SkyCoords
+    coord_asu = SkyCoord(ra=asu.iloc[2:]['RAJ2000'], dec=asu.iloc[2:]['DEJ2000'], unit=('hourangle', 'deg'))
+    coord_fri = SkyCoord(ra=fri_c[:, 0], dec=fri_c[:, 1], unit=('hourangle', 'deg'))
+    coord_frii = SkyCoord(ra=frii_c[:, 0], dec=frii_c[:, 1], unit=('deg', 'deg'))
+    #coord_first = SkyCoord(ra=first.iloc[ind_first]['RA'], dec=first.iloc[ind_first]['Dec'], unit=('hourangle', 'deg'))
+    
+    #first_plot = SkyCoord(ra=first['RA'], dec=first['Dec'], unit=("hourangle", "deg"))
+
+    #plt.scatter(first_plot.ra.deg, first_plot.dec.deg, label="first")
+    #plt.scatter(coord_asu.ra.deg, coord_asu.dec.deg, label="asu", alpha=0.5)
+    #plt.scatter(coord_fri.ra.deg, coord_fri.dec.deg, label="fri")
+    #plt.scatter(coord_frii.ra.deg, coord_frii.dec.deg, label="frii", alpha=0.5)
+    #plt.legend()
+    #plt.show()
+
+    # Finding nearest FRI's and FRIIs to each object in ASU
+    idx_i, sep_i, _i = coord_fri.match_to_catalog_sky(coord_asu)
+    idx_ii, sep_ii, _ii = coord_frii.match_to_catalog_sky(coord_asu)
+
+    # For sources within our set threshold, we assume that they are the same object.
+    fri_in_asu = (sep_i <= 5*u.arcsec)
+    frii_in_asu = (sep_ii <= 5*u.arcsec)
+
+    # Extacting FR images that are in asu
+    asu_fri = fri_images[fri_in_asu]
+    asu_frii = frii_images[frii_in_asu]
+
+    leftover_fri = fri_images[np.invert(fri_in_asu)]
```

Comment:
I assume `invert` flips the bool? You can also use not (`~`) if it's a boolean array.